### PR TITLE
Chat reactions: desktop integration

### PR DESCRIPTION
**Pull request by buoyad**
Full integration on desktop. Also adds a feature flag for it. I ran into issues with circular dependencies in storybook, which is why `ReactButtonWithTooltip` is factored out into a separate file. r? @keybase/react-hackers

### CODE REVIEW

File: shared/chat/conversation/list-area/normal/index.desktop.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // @flow
+/* eslint-env browser */
```
Comment (chrisnojima) [inline shared/chat/conversation/list-area/normal/index.desktop.js]: why is this necessary?

Comment (buoyad) [reply shared/chat/conversation/list-area/normal/index.desktop.js]: eslint was marking `MouseEvent` as undefined

Comment (chrisnojima) [reply shared/chat/conversation/list-area/normal/index.desktop.js]: you're using the wrong type: https://github.com/facebook/flow/blob/v0.76.0/lib/react-dom.js#L193

Comment (buoyad) [reply shared/chat/conversation/list-area/normal/index.desktop.js]: oh it's actually complaining about `HTMLInputElement` and `HTMLTextAreaElement` as well
